### PR TITLE
[Merged by Bors] - Add compiler script of torch model to torchscript model

### PR DESCRIPTION
**Pull request by xidianwym**
Signed-off-by: Wang Yiming <wang.yiming6@zte.com.cn>

### CODE REVIEW

**Review by shiyingjin — CHANGES_REQUESTED** [review]
重新rebase代码

**Review by NaNAGISaSA — CHANGES_REQUESTED** [review]

File: model_compiler/src/model_compiler/compilers/torch_model_file_to_torchscript_model.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        shape.insert(0, config.max_batch_size)
+        dummy_inputs.append(torch.ones(shape, dtype=config.data_type))   # pylint: disable=no-member
+
+    if source.script_path:
+        model_module = utilities.load_module(source.script_path, 'TheModelClass')
+        model = model_module.TheModelClass()
+        model.load_state_dict(torch.load(source.model_path, map_location='cpu'))
+    else:
+        model = torch.load(source.model_path, map_location='cpu')
+    temp_path = TemporaryDirectory()
```
[the remark this answers is not in the document]
Comment (xidianwym) [reply model_compiler/src/model_compiler/compilers/torch_model_file_to_torchscript_model.py:31]: 利用TemporaryDirectory构建的临时目录，在执行完上下文之后，会自动删除，不需要增加显示的删除代码

**Review by shiyingjin — APPROVED** [review]

File: model_compiler/src/model_compiler/compilers/torch_model_file_to_torchscript_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NaNAGISaSA) [reply model_compiler/src/model_compiler/compilers/torch_model_file_to_torchscript_model.py:31]: see how to use the context manager: [link](https://docs.python.org/3/library/tempfile.html#examples)

**Review by NaNAGISaSA — CHANGES_REQUESTED** [review]

File: model_compiler/src/model_compiler/models/data_type.py [hunk 2]
```diff
@@ -123,20 +123,20 @@ def from_torch_data_type(type_str):
...
+            torch.complex64: DataType.COMPLEX64,
+            torch.complex128: DataType.COMPLEX128,
+            torch.float16: DataType.FLOAT16,
+            torch.bfloat16: DataType.BFLOAT16,
+            torch.uint8: DataType.UINT8,
+            torch.int8: DataType.INT8,
+            torch.int16: DataType.INT16,
+            torch.int32: DataType.INT32,
+            torch.int64: DataType.INT64,
+            torch.bool: DataType.BOOL,
```
Comment (NaNAGISaSA) [inline model_compiler/src/model_compiler/models/data_type.py:137]: the reason for this change?

File: model_compiler/tests/model_compiler/compilers/test_torch_model_file_to_onnx_model.py [hunk 3]
```diff
@@ -18,27 +19,27 @@ class ConfigTestCase(TestCase):
     def test_from_json(self):
         self.assertEqual(Config.from_json({'input_names': ['input'],
                                            'input_shapes': [1, 2, 3, 4],
-                                           'data_type': 'FLOAT16',
+                                           'data_type': torch.float,     # pylint: disable=no-member
```
Comment (NaNAGISaSA) [inline model_compiler/tests/model_compiler/compilers/test_torch_model_file_to_onnx_model.py:22]: I think json config and env are common configs, it should not be related to frameworks like torch, tf etc.

**Review by NaNAGISaSA — CHANGES_REQUESTED** [review]

**Pull request closed without merging**